src/Tools/dotnet-monitor/Auth/GeneratedApiKey.cs

```diff
@@ -23,15 +27,25 @@ private GeneratedApiKey(string monitorApiKey, string hashAlgorithm, string hashV
 
         public static GeneratedApiKey Create()
         {
-            return GeneratedApiKey.Create("SHA256");
+            return GeneratedApiKey.Create(DefaultKeyLength, DefaultHashAlgorithm);
         }
 
-        private static GeneratedApiKey Create(string hashAlgorithm)
+        public static GeneratedApiKey Create(int keyLength, string hashAlgorithm)
         {
+            if (!HashAlgorithmChecker.IsAllowedAlgorithm(hashAlgorithm))
+            {
+                throw new ArgumentOutOfRangeException(nameof(hashAlgorithm));
+            }
+
+            if (keyLength < ApiKeyAuthenticationHandler.ApiKeyByteMinLength || keyLength > ApiKeyAuthenticationHandler.ApiKeyByteMaxLength)
+            {
+                throw new ArgumentOutOfRangeException(nameof(keyLength));
+            }
+
             using RandomNumberGenerator rng = RandomNumberGenerator.Create();
             using HashAlgorithm hasher = System.Security.Cryptography.HashAlgorithm.Create(hashAlgorithm);
 
-            byte[] secret = new byte[ApiKeyAuthenticationHandler.ApiKeyNumBytes];
+            byte[] secret = new byte[keyLength];
```

Comment:
Should the key length be at least as long as the hash length?

Reply:
I think it's simpler to have the "more than 32 bytes" restriction be constant and let the user pick the algorithm. While I don't think it makes sense to have a 32 byte key and a 64 byte hash output, I don't think it has a meaningful security implication so I'm leaning towards the simpler restriction.